imcache_test.go

```diff
@@ -679,24 +679,30 @@ type evictionCallbackMock struct {
 
 func (m *evictionCallbackMock) Callback(key string, val interface{}, reason EvictionReason) {
 	m.mu.Lock()
-	m.calls = append(m.calls, evictionCallbackCall{key, val, reason})
+	m.calls = append(m.calls, evictionCallbackCall{key: key, val: val, reason: reason})
 	m.mu.Unlock()
 }
 
-func (m *evictionCallbackMock) HasBeenCalledWith(t *testing.T, key string, val interface{}, reason EvictionReason) {
+func (m *evictionCallbackMock) HasEventuallyBeenCalledWith(t *testing.T, key string, val interface{}, reason EvictionReason) {
 	t.Helper()
-	m.mu.Lock()
-	calls := make([]evictionCallbackCall, 0, len(m.calls))
-	for _, c := range m.calls {
-		if c.key == key {
-			calls = append(calls, c)
+	initialBackoff := 20 * time.Millisecond
+	backoffCoefficient := 2
+	for i := 0; i < 5; i++ {
+		m.mu.Lock()
+		calls := make([]evictionCallbackCall, 0, len(m.calls))
+		for _, c := range m.calls {
```

Comment:
This can be smarter. Since calls are append-only within a single test we can start the iteration from where we left.